### PR TITLE
Fixed #30636 -- Fixed options ordering when cloning test database on MySQL.

### PR DESCRIPTION
**Pull request by fallen**

### CODE REVIEW

File: django/db/backends/mysql/creation.py [hunk 1]
```diff
@@ -56,8 +56,10 @@ def _clone_test_db(self, suffix, verbosity, keepdb=False):
 
     def _clone_db(self, source_database_name, target_database_name):
         dump_args = DatabaseClient.settings_to_cmd_args(self.connection.settings_dict)[1:]
-        dump_args[-1] = source_database_name
-        dump_cmd = ['mysqldump', '--routines', '--events'] + dump_args
+        if dump_args[-1] == source_database_name:
+            dump_args.pop()
+        dump_args.extend(['--routines', '--events', source_database_name])
+        dump_cmd = ['mysqldump'] + dump_args
```
[the remark this answers is not in the document]
Comment (fallen) [reply django/db/backends/mysql/creation.py]: IIUC you propose to only keep the 1st element of dump_args?
What happens if it contains other necessary arguments?

[the remark this answers is not in the document]
Comment (felixxm) [reply django/db/backends/mysql/creation.py]: Sorry typo:
```python
dump_cmd = ['mysqldump', *dump_args[:-1], '--routines', '--events', source_database_name]
```

[the remark this answers is not in the document]
Comment (felixxm) [reply django/db/backends/mysql/creation.py]: My proposition keeps all arguments.

[the remark this answers is not in the document]
Comment (fallen) [reply django/db/backends/mysql/creation.py]: This seems to unconditionally drop the last element of dump_args.
And indeed  self.connection.settings_dict["NAME"] seems to be set in all cases by https://github.com/django/django/blob/master/django/db/backends/base/creation.py#L62
Let's go for your version.
Thanks!

**Comment by fallen** [comment]
For the test, I would need time to write it as I am not used to Mock stuff at all, and this use-case seems rather advanced.
If you want to write the test yourself please feel free to do it.
If not, I'll do it but it will take a bit of time :)

**Comment by felixxm** [comment]
@fallen I've provided test in [comment](https://github.com/django/django/pull/11567#pullrequestreview-261637553). I can add it to this patch, np.

File: django/db/backends/mysql/creation.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (felixxm) [reply django/db/backends/mysql/creation.py]: Previously we always changed the last option to `source_database_name`, see
```python
dump_args[-1] = source_database_name
```
so it keeps this behavior.

[the remark this answers is not in the document]
Comment (fallen) [reply django/db/backends/mysql/creation.py]: Ok, I'll post an update to the patch with your modification later in the evening.

[the remark this answers is not in the document]
Comment (fallen) [reply django/db/backends/mysql/creation.py]: forced push to reflect this discussion.

**Comment by felixxm** [comment]
@fallen Thanks! I added test.

**Comment by fallen** [comment]
Thanks a lot!

**Pull request merged**